pkg/nmi/server/endpointpolicy.go

```diff
@@ -136,7 +137,32 @@ func deleteEndpointPolicy(endpoint *v1.HNSEndpoint, metadataIP string) error {
 }
 
 func callHcnProxyAgent(req msg.HNSRequest) ([]byte, error) {
+	retryCount := 1
+	maxRetryCount := 4
+	var sleepFactor time.Duration = 1
+
 	klog.Info("Calling HNS Agent")
+
+	for {
+		response, err := callHcnProxyAgentInternal(req)
+		if err != nil {
+			if retryCount > maxRetryCount {
+				klog.Info("Calling HNS Agent failed after all retries, giving up")
```

Comment:
This should be klog.Infof() or klog.Errorf()

Reply:
Wait, this shout not be f right, its has not format.

---
In reply to: [439687676](https://github.com/bowu24/aad-pod-identity/pull/5#discussion_r439687676) [](ancestors = 439687676)